src/frontends/ir/src/ir_deserializer.cpp

```diff
@@ -512,7 +512,17 @@ std::shared_ptr<ov::Model> ov::XmlDeserializer::parse_function(const pugi::xml_n
         }
 
         if (const auto& sink = std::dynamic_pointer_cast<ov::op::Sink>(node)) {
-            func_nodes.sinks.emplace_back(sink);
+            auto subgraph_op = std::dynamic_pointer_cast<ov::op::util::MultiSubGraphOp>(node);
+            if (subgraph_op) {
+                for (const auto& body_model : subgraph_op->get_functions()) {
+                    if (body_model->get_sinks().size()) {
+                        func_nodes.sinks.emplace_back(sink);
+                        break;
+                    }
+                }
+            } else {
+                func_nodes.sinks.emplace_back(sink);
```

Comment:
I think here we should check if it casts to `KeepInGraphOp` using `as_type_ptr<...>`